internal/cri/server/cni_conf_syncer.go

```diff
@@ -97,13 +98,16 @@ func (syncer *cniNetConfSyncer) syncLoop() error {
 			}
 			log.L.Debugf("receiving change event from cni conf dir: %s", event)
 
-			lerr := syncer.netPlugin.Load(syncer.loadOpts...)
-			if lerr != nil {
-				log.L.WithError(lerr).
-					Errorf("failed to reload cni configuration after receiving fs change event(%s)", event)
+			if isValidFile(event.Name) {
+				lerr := syncer.netPlugin.Load(syncer.loadOpts...)
+				if lerr != nil {
+					log.L.WithError(lerr).
+						Errorf("failed to reload cni configuration after receiving fs change event(%s)", event)
+				}
+				syncer.updateLastStatus(lerr)
+			} else {
+				log.L.Debugf("not recognized file extension: %s", event)
```

Comment:
let's do this if !isValidFileExtension(event.Name) up on line 96 as another debug ignore check before continuing ...

Reply:
hmm.. yeah problem here if someone renames a valid file to a file.extension.temp file we may have an invalid directory now..

Reply:
we can "skip" create when it's not a valid extension..  and we can force a retest when it's fsnotify.Rename .. need to think all the cases out.. 

Cheers

Reply:
Is this what you were proposing? 

`if !isValidFileExtension(event.Name) || event.Has(fsnotify.Chmod) || event.Has(fsnotify.Create) {
				log.L.Debugf("ignore event from cni conf dir: %s", event)
				continue
			}`